vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -493,6 +494,21 @@ public Component getControls() {
       alreadyTaken.add(p.side);
     }
 
+     /*
+     Scan for excluded sides based on translation of property VassalExcludeSide_<Side>
+     add such to the alreadyTaken list
+     Note that <observer> can't be excluded as it is added later
+     Usage:
+      For scenario-specific exclusion, define VassalExcludeSide_ properties in Scenario Options
+      For dynamic exclusion, define as module Global Properties
+    */
+    for (int i = 0; i < sides.size(); i++) {
+      final String s = sides.get(i);
+      if (!alreadyTaken.contains(s) && StringUtils.equalsIgnoreCase((String) GameModule.getGameModule().getProperty("VassalExcludeSide_" + (untranslatedSides == null ? s : untranslatedSides[i])), "true")) {
```

Comment:
Don't call `GameModule.getGameModule()` each time through the loop. Get the `GameModule` once, before entering the loop.

Reply:
resolving in next push.